### ISSUE TITLE
Don't warn for unprotected array appending

### ISSUE THREAD

**Issue by DamienCassou** [koalaman/shellcheck#234]
``` bash
declare -a files
files+=($(basename "$file"))
files+=( $file )
```

The 2 last lines are problematic but shellcheck won't tell me.

**Comment by koalaman** [comment]
This has been finally been fixed in 3c75674b with the introduction of some array specific warnings!
